### PR TITLE
Ready for testing 🧪 Multi-policy training support

### PR DESCRIPTION
**Pull request by Ivan-267**
Adds support for training multiple policies with Rllib. 

Plugin PR: https://github.com/edbeeching/godot_rl_agents_plugin/pull/40
Example env PR: https://github.com/edbeeching/godot_rl_agents_examples/pull/30

TODO:
- [x] Fix multiple obs spaces case (should be fixed now but still needs more checking, onnx export doesn't currently work with multiple discrete obs spaces, I think it's configured for a single space, will need to check this at some point as well).

### CODE REVIEW

**Comment by Ivan-267** [comment]
I've done a little testing with some of my previous envs and Jumper Hard with older plugin version, and it seemed to work with both multiagent set to `false` (possible that number of envs per worker should be adjusted manually), or `true` (not intended for single agent envs due to individual agents being inactivated in rllib after `done = true`, but should not cause errors due to the compatibility code in `GDRLPettingZooWrapper`). SB3 seems to work properly after these changes, but I only tested in on a modified version of the multi-agent env for now (made into a single agent compatible version). Further testing is always welcome, especially on Linux and Sample Factory.

LSTM/Attention wrappers work (they show a deprecated warning so possibly accessing them might be different when newer versions of rllib come out), but for exporting we can't use them yet since as the state data wouldn't be fed in.

One thing I found that doesn't work well is enabling some exploration noises with PPO, one that worked was RE3 with Tensorflow rather than Torch set. Curiosity needs discrete or multidiscrete actions, but didn't seem to work when I switched the env to discrete actions. I think it might be related to the tuple action space, it might not be supported in some of the exploration codes.

File: godot_rl/wrappers/ray_wrapper.py
Comment (edbeeching) [inline godot_rl/wrappers/ray_wrapper.py:29]: Can all the options in `__init__` such as `env_path=None` etc be removed now as they seem to be stored in config?

**Review by edbeeching — APPROVED** [review]
LGTM pending review of other PRs

File: examples/rllib_example.py
Comment (edbeeching) [inline examples/rllib_example.py:21]: I think this should be `examples/rllib_config.ymal`

File: godot_rl/wrappers/ray_wrapper.py
Comment (Ivan-267) [reply godot_rl/wrappers/ray_wrapper.py:29]: I think we only need e.g. as used in rllib example:
```return RayVectorGodotEnv(config=env_config, port=port, seed=seed)```

Perhaps even `seed` and `port` can be removed, but maybe it's more clear that they are dynamic options this way (we don't normally use a fixed port and seed for multiple envs).

We would need to change the register env also to make sure tests aren't broken.

I will check this later, try to remove them and see if there are any issues.

File: examples/rllib_example.py
Comment (Ivan-267) [reply examples/rllib_example.py:21]: I usually call the example from within the examples folder, so the default was based on my usage. If calling from GDRL repository directly then it should be changed. 

If someone installs GDRL using `pip install` and then just downloads the example file and config file, they might not have the entire repository, but I'm not sure how common this is.

I leave this up to you, I can definitely change the default.

**Review by edbeeching — APPROVED** [review]

**Pull request merged**
